### ISSUE TITLE
Constraints not user properly

### ISSUE THREAD

**Issue by gitbuda** [memgraph/memgraph#1679]
Constraints do not avoid the same nodes being ingested several times.

**Comment by DavIvek** [comment]
I wasn't able to reproduce this with memgraph 2.14.1 using `IN_MEMORY_TRANSACTIONAL` .

**Comment by katarinasupe** [comment]
@DavIvek This was reported by one user without steps to reproduce. I haven't heard such reports from other sides, so I wouldn't investigate further until there are valid concerns about this and steps to reproduce.

**Comment by Josipmrden** [comment]
> ```
> STORAGE MODE IN_MEMORY_ANALYTICAL;
> CREATE CONSTRAINT ON (e:TEMP) ASSERT e.id_entity IS UNIQUE;
> CREATE (e:TEMP {id_entity: 1});
> CREATE (e:TEMP {id_entity: 1});
> MATCH (n:TEMP) RETURN n;
> ```
> 
> Running these one by one in memgraph lab returns two entities with the same id_entity:
> 
> ```
> {"n":{"id":81858,"labels":["TEMP"],"properties":{"id_entity":1},"type":"node"}}
> {"n":{"id":81859,"labels":["TEMP"],"properties":{"id_entity":1},"type":"node"}}
> ```
> 
> FWIW, I tried to run the same but setting the constraint while still in transactional mode, but this still led to two entities being created.
> 
> ```
> MATCH (n:TEMP) DETACH DELETE n;
> STORAGE MODE IN_MEMORY_TRANSACTIONAL;
> CREATE CONSTRAINT ON (e:TEMP) ASSERT e.id_entity IS UNIQUE;
> STORAGE MODE IN_MEMORY_ANALYTICAL;
> CREATE (e:TEMP {id_entity: 1});
> CREATE (e:TEMP {id_entity: 1});
> MATCH (n:TEMP) RETURN n;
> ```

Thanks for reproducing! We'll look into it

**Comment by gitbuda** [comment]
The constraint validation has to be on the storage layer.
When initially discussed, `IN_MEMORY_ANALYTICAL` doesn't provide any ACID guarantees (except snapshotting/backup data); an option here is to disable constraint creation completely because constraint checking introduces additional overhead.

**Comment by Danferno** [comment]
I don't come from a database world so I'm not terribly familiar with the ACID term, but would an intermediate step be to update the [documentation here](https://memgraph.com/docs/fundamentals/storage-memory-usage#switch-storage-modes) to (addition in bold)

In the practical sense, not having ACID properties leads to the following implications:

    Atomicity - if your write transaction fails for any reason, the changes are not rolled back.
    Consistency - database consistency is not guaranteed as transactions may fail and leave the database in an inconsistent state. **Any uniqueness or existence (?) constraint checks are skipped. **
    Isolation - if you have a long-running write transaction, other transactions can see the changes of ongoing transactions even if changes are not committed.
    Durability - Memgraph does not create WAL files and periodic snapshots in the analytical storage mode. They need to be created manually.

**Comment by antejavor** [comment]
Yeah, @Danferno agreed it makes it more understandable. I will improve the docs.
